pymongo/mongo_client.py

```diff
@@ -368,45 +369,57 @@ def __init__(
+            your application.
+          - `tlsAllowInvalidCertificates`: (boolean) Specifies whether a
+            certificate is required from the other side of the connection, and
+            whether it will be validated if provided. Certificates are ignored
+            if ``tlsAllowInvalidCertificates=True``. Certificates are required
+            and validated if ``tlsAllowInvalidCertificates=False``. If the
+            value of this parameter is ``False`` and a value is not provided
+            for ``tlsCAFile``, PyMongo will attempt to load system provided CA
+            certificates. If the python version in use does not support loading
+            system CA certificates then the ``tlsCAFile`` parameter must point
+            to a file of CA certificates. ``tlsAllowInvalidCertificates=False``
+            implies ``tls=True``. Defaults to ``False``.
+          - `tlsAllowInvalidHostnames`: (boolean) If ``False`` and
+            `tlsAllowInvalidCertificates` is ``False``, enables hostname
+            verification using the :func:`~ssl.match_hostname` function from
+            python's :mod:`~ssl` module. ``tlsAllowInvalidHostnames=False``
+            implies ``tls=True``. Defaults to ``False``. Think very carefully
+            before setting this to ``True`` as that could make your application
+            vulnerable to man-in-the-middle attacks.
+          - `tlsCAFile`: The file containing a single or a bundle of
             "certification authority" certificates, which are used to validate
             certificates passed from the other end of the connection.
-            Implies ``ssl=True``. Defaults to ``None``.
+            Implies ``tls=True``. Defaults to ``None``.
+          - `tlsCertificateKeyFile`: The client certificate file or the client
+            private key file used to identify the local connection against
+            mongod. In the case that they are both needed, the files can either
+            be concatenated and passed to this option, or they can be passed
+            separately using the ``tlsCertificateKeyFile`` and ``ssl_keyfile``
+            options respectively. Implies ``tls=True``. Defaults to ``None``.
+          - `ssl_certfile`: Alias for ``tlsCertificateKeyFile``.
+          - `ssl_keyfile`: The private keyfile used to identify the local
+            connection against mongod. Can be omitted if the keyfile is
+            included with the ``tlsCertificateKeyFile``. Implies ``tls=True``.
+            Defaults to ``None``.
+          - `tlsCertificateKeyFilePassword`: The password or passph rase for
+            decrypting the private key in ``tlsCertificateKeyFile`` or
+            ``ssl_keyfile``. Only necessary if the private key is encrypted.
+            Only supported by python 2.7.9+ (pypy 2.5.1+) and 3.3+. Defaults
+            to ``None``.
           - `ssl_crlfile`: The path to a PEM or DER formatted certificate
```

Comment:
Didn't we add a new tlsCRLFile option?

Reply:
Nope. Only the Python driver implements the CRLFile option for some reason.

Reply:
Sorry, I did add a `tlsCRLFile` option but forgot to document it. I'll add it in the next commits.